lib/fileHandler/index.js

```diff
@@ -47,19 +47,57 @@ const DEFAULT_UPLOAD_OPTIONS = {
 	queueSize: 4 //Ensures at least up to 1/4 of the RAM is utilised
 };
 
-const REQUIRED_FILE_PROPS = ['name', 'url', 'mime_type', 'expires']
+const REQUIRED_FILE_PROPS = [ 'name', 'url', 'mime_type', 'expires' ];
+const checkIfStr = (val) => { return typeof val === 'string'; };
+
+const executeTypeValidation = (key, value) => {
+	const fieldValidationMap = {
+		name: (value) => { return checkIfStr(value); },
+		url: (value) => { return checkIfStr(value); },
+		mime_type: (value) => { return checkIfStr(value); },
+		expires: (value) => { return typeof value === 'number'; }
+	};
 
-const isObjValid = fileObj => {
-    const fileObjKeyArr = _.keys(fileObj)
-    return REQUIRED_FILE_PROPS.every(reqField => fileObjKeyArr.includes(reqField))
-}
+	const isValidType = fieldValidationMap[key](value);
+    
+	if (!isValidType) {
+		return key;
+	}
+};
+
+const runFileValidation = (fileObj) => {
+	const fileObjKeyArr = _.keys(fileObj);
+	const isValid =  REQUIRED_FILE_PROPS.every((reqField) => { return fileObjKeyArr.includes(reqField); });
```

Comment:
Given `.every` and `.reduce` are both looping through the entire REQUIRED_FILE_PROPS array, you should just be able to combine the logic of both checks into a single loop.

Also, to aid with this, instead of a `executeTypeValidation` function with a `fieldValidationMap` object, (which would need to be kept in sync with the `REQUIRED_FILE_PROPS` array if that were ever to updated), it'd be better off to modify `REQUIRED_FILE_PROPS` into array of objects. Each object would have say `key` and `validator` property for example, with  `validator` being a function, e.g.
```js
const REQUIRED_FILE_PROPS = [
	{
		key: 'name',
		validator: isString
	},
	...etc
]
```
This way you can consolidate both loops and check both the key and value in on go.

Reply:
ye makes sense, does simply things a fair bit